packages/react/src/components/TearSheet/TearSheetWrapper.jsx

```diff
@@ -48,6 +49,24 @@ const TearSheetWrapper = ({ isOpen, className, onCloseAllTearSheets, children })
     )
   );
 
+  const tearSheetsNodes = containers.reduce(
+    (acc, curr) => [...acc, curr?.current?.children[0]],
+    []
+  );
+
+  useEffect(() => {
+    const tearSheetHeaderHeight = Math.round(
+      tearSheetsNodes?.[activeTearSheetIdx]?.children?.[0].getBoundingClientRect().height
```

Comment:
This probably makes more sense as a useLayoutEffect since you're calling getBoundingClientRect for measurements. https://kentcdodds.com/blog/useeffect-vs-uselayouteffect